src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -0,0 +1,182 @@
+    [ExtensionOrder(Before = QuickInfoProviderNames.Semantic)]
+    internal class CSharpDiagnosticAnalyzerQuickInfoProvider : CommonQuickInfoProvider
+    {
+        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpDiagnosticAnalyzerQuickInfoProvider(IDiagnosticAnalyzerService diagnosticAnalyzerService)
+        {
+            _diagnosticAnalyzerService = diagnosticAnalyzerService;
+        }
+
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
+            Document document,
+            SyntaxToken token,
+            CancellationToken cancellationToken)
+        {
+            return GetQuickinfoForPragmaWarning(document, token) ??
+                (await GetQuickInfoForSuppressMessageAttributeAsync(document, token, cancellationToken).ConfigureAwait(false));
+        }
+
+        private QuickInfoItem? GetQuickinfoForPragmaWarning(Document document, SyntaxToken token)
+        {
+            var errorCode = token.Parent switch
+            {
+                PragmaWarningDirectiveTriviaSyntax directive
+                    => token.IsKind(SyntaxKind.EndOfDirectiveToken)
+                        ? directive.ErrorCodes.LastOrDefault() as IdentifierNameSyntax
+                        : directive.ErrorCodes.FirstOrDefault() as IdentifierNameSyntax,
+                IdentifierNameSyntax { Parent: PragmaWarningDirectiveTriviaSyntax _ } identifier
+                    => identifier,
+                _ => null,
+            };
+
+            if (errorCode != null)
+            {
+                return GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(document, errorCode.Identifier.ValueText, errorCode.Span);
+            }
+
+            return null;
+        }
+
+        private async Task<QuickInfoItem?> GetQuickInfoForSuppressMessageAttributeAsync(
+            Document document,
+            SyntaxToken token,
+            CancellationToken cancellationToken)
+        {
+            // SuppressMessageAttribute docs 
+            // https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute
+            var suppressMessageCheckIdArgument = token.GetAncestor<AttributeArgumentSyntax>() switch
+            {
+                AttributeArgumentSyntax
+                {
+                    Parent: AttributeArgumentListSyntax
+                    {
+                        Arguments: var arguments,
+                        Parent: AttributeSyntax
+                        {
+                            Name: var attributeName
+                        } _
+                    } _
+                } argument when
+                    attributeName.IsSuppressMessageAttribute() &&
```

Comment:
This is a syntactical test (Name=="SupressMessage") only. It does not take type aliases into account. This is documented by a test, but I can change the behavior if desired.